pkg/certproviderutil/types.go

```diff
@@ -0,0 +1,41 @@
+// Package certproviderutil implements utility functions to manage different certificate managers/providers.
+package certproviderutil
+
+import (
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/openservicemesh/osm/pkg/configurator"
+	"github.com/openservicemesh/osm/pkg/logger"
+)
+
+var log = logger.New("cert-provider-util")
+
+// CertProviderUtil is a type that implements utility functions associated with certificate providers
+type CertProviderUtil struct {
+	KubeClient            kubernetes.Interface
+	KubeConfig            *rest.Config
+	Cfg                   configurator.Configurator
+	CertProviderKind      string
+	CertProviderNamespace string
+	CABundleSecretName    string
+
```

Comment:
I'd even add TresorOptions here -- a) to make it obvious that this lists all 3 available cert issuers and b) gives us an opportunity to centralize Tresor config here.

Reply:
Done.